content/thursday/exercises_Bayesian_regularization.qmd

```diff
@@ -0,0 +1,449 @@
+
+### Comparing results across different priors
+Now that we have a good idea of the different shrinkage prior distributions, we can run the analysis for different prior specifications and compare the outcome of interest, in our case: the selection of variables to predict abalone age. Run the analysis with Student's t, the lasso, and horseshoe priors specified above. You can also add one or more of the priors you investigated yourself. Compare the amount of shrinkage, for example by looking at the estimated regression coefficients or the full posteriors, as well as the number of selected variables based on the 95% credible intervals. Do the results differ across different shrinkage priors and are the differences in line with your intuition regarding the shrinkage behaviors of the priors?
+
+
+```{r}
+#| message: false
+#| eval: false
+#| code-fold: true
+#| code-summary: "Show the code"
+fit_t <- brm(Rings ~ ., data = train,
+             prior = set_prior("student_t(3, 0, 0.1)", class = "b"))
+# Indicates no relevant variables (like the ridge)
+
+fit_lasso <- brm(Rings ~ ., data = train,
+             prior = set_prior(lasso(df = 1, scale = 1), class = "b"))
+# Selects SexM, Shell_weight, and Shucked_weight
+
+fit_hs <- brm(Rings ~ ., data = train,
+             prior = set_prior(horseshoe(df = 1, scale_global = 1,
+                                     df_global = 1, scale_slab = 2,
+                                     df_slab = 4, par_ratio = NULL,
+                                     autoscale = TRUE), class = "b"))
+# Selects SexM and Shell_weight
+
+# Visualize differences
+draws_ridge <- as_draws_df(fit_ridge)
+draws_ridge$prior <- "ridge"
+draws_t <- as_draws_df(fit_t)
+draws_t$prior <- "t"
+draws_lasso <- as_draws_df(fit_lasso)
+draws_lasso$prior <- "lasso"
+draws_hs <- as_draws_df(fit_hs)
+draws_hs$prior <- "hs"
+
+sel <- c("b_SexI", "b_SexM", "b_Length", "b_Diameter", "b_Height", 
+          "b_Whole_weight", "b_Shucked_weight", "b_Viscera_weight",
+          "b_Shell_weight", "prior")
+draws <- rbind.data.frame(draws_ridge[, sel],
+                          draws_t[, sel], 
+                          draws_lasso[, sel], 
+                          draws_hs[, sel])
+draws$id <- 1:nrow(draws)
+draws_long <- gather(draws, 
+                     key = "parameter",
+                     value = "value",
+                     -prior)
+
+ggplot(draws_long, aes(value, colour=prior)) +
+  geom_density() +
+  facet_wrap(~ parameter, scales = "free")
+# We see very similar results for the ridge and Student's t; these priors shrink coefficients heavily, with no escape for larger coefficients. The lasso and horseshoe also show similar results and less shrinkage for some parameters, as we would expect based on the prior plots.
+```
+
```

Comment:
Maybe add something with a clear note about divergent transitions and low ESS warnings related to the HS prior model. What should they do? 

You can create a callout section like this:

::: callout-note
something about the divergent transitions etc, 
:::